osu.Game/Screens/Menu/IntroWelcome.cs

```diff
@@ -0,0 +1,163 @@
+                                {
+                                    visualizer = new LogoVisualisation
+                                    {
+                                        RelativeSizeAxes = Axes.Both,
+                                        Anchor = Anchor.Centre,
+                                        Origin = Anchor.Centre,
+                                        Alpha = 0.5f,
+                                        IsIntro = true,
+                                        AccentColour = Color4.DarkBlue,
+                                        Size = new Vector2(0.96f)
+                                    },
+                                    circleContainer = new Container
+                                    {
+                                        AutoSizeAxes = Axes.Both,
+                                        Children = new Drawable[]
+                                        {
+                                            blackCircle = new Circle
+                                            {
+                                                Anchor = Anchor.Centre,
+                                                Origin = Anchor.Centre,
+                                                Size = new Vector2(480),
+                                                Colour = Color4.Black
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    welcomeText = new Sprite
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Scale = new Vector2(0.3f),
+                        Width = 750,
+                        Height = 78,
+                        Alpha = 0,
+                        Texture = textures.Get(@"Welcome/welcome_text@2x")
+                    },
+                };
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                double remainingTime() => delay_step_two - TransformDelay;
+
+                using (BeginDelayedSequence(0, true))
+                {
+                    welcomeText.ResizeHeightTo(welcomeText.Height * 2, 500, Easing.In);
+                    welcomeText.FadeIn(remainingTime());
+                    welcomeText.ScaleTo(welcomeText.Scale + new Vector2(0.1f), remainingTime(), Easing.Out).OnComplete(_ =>
+                    {
+                        elementContainer.Remove(visualizer);
+                        circleContainer.Remove(blackCircle);
+                        elementContainer.Remove(circleContainer);
+                        Remove(welcomeText);
+                        visualizer.Dispose();
+                        blackCircle.Dispose();
+                        welcomeText.Dispose();
```

Comment:
I'm not overly familiar with drawable lifetime stuff, but isn't `Expire()` what you actually want here? If you expire the elements they should automatically get cleaned up from the containers. [There's even a framework-side test case for that.](https://github.com/ppy/osu-framework/blob/2637ba3af75e5ef0d3e05388309e153e457f489c/osu.Framework.Tests/Containers/TestSceneContainerState.cs#L222-L239)